src/main/java/burp/CustomHighlight.java

```diff
@@ -53,6 +61,17 @@ class CustomHttpResponseEditor implements ExtensionProvidedHttpResponseEditor, P
         // Burp Suite uses apparently a modified JTextArea, so we can cast to to one
         this.textArea = (JTextArea) scrollPane.getViewport().getView();
 
+        setBaselineButton = new JButton("Set As Diff Base");
+        setBaselineButton.addActionListener(e -> {
+            HttpResponse response = getResponse();
+            Optional<HttpResponse> responseAsOptional = (response != null) ? Optional.of(response) : Optional.empty();
+            BaseResponse.setBaseResponse(responseAsOptional);
```

Comment:
You should be able to do:

```suggestion
            Optional<HttpResponse> response = Optional.ofNullable(getResponse());
            BaseResponse.setBaseResponse(response);
```

Reply:
Implemented